distribution/std-lib/Base/src/Time/Time.enso

```diff
@@ -0,0 +1,177 @@
+from Base import all
+import Base.Time.Date
+import Base.Time.Duration
+import Base.Time.Time_Of_Day
+import Base.Time.Zone
+
+polyglot java import java.time.ZonedDateTime
+polyglot java import java.time.format.DateTimeFormatter
+polyglot java import org.enso.base.Time_Utils
+
+type Time
+
+    ## A date-time with a time-zone in the ISO-8601 calendar system, such as
+       "2007-12-03T10:15:30+01:00 Europe/Paris".
+
+       Time is a representation of a date-time with a time-zone. This class
+       stores all date and time fields, to a precision of nanoseconds, and a
+       time-zone, with a zone offset used to handle ambiguous local
+       date-times.
+
+       For example, the value "2nd October 2007 at 13:45.30.123456789 +02:00 in
+       the Europe/Paris time-zone" can be stored as Time.
+    type Time internal_zoned_date_time
+
+    ## Get the year field.
+    get_year : Integer
+    get_year = this . internal_zoned_date_time . getYear []
+
+    ## Get the month of year field from 1 to 12.
+    get_month : Integer
+    get_month = this . internal_zoned_date_time . getMonthValue []
+
+    ## Get the day of month field.
+    get_day : Integer
+    get_day = this . internal_zoned_date_time . getDayOfMonth []
+
+    ## Get the hour of day field.
+    get_hour : Integer
+    get_hour = this . internal_zoned_date_time . getHour []
+
+    ## Get the minute of hour field.
+    get_minute : Integer
+    get_minute = this . internal_zoned_date_time . getMinute []
+
+    ## Get the second of minute field
+    get_second : Integer
+    get_second = this . internal_zoned_date_time . getSecond []
+
+    ## Get the nano-of-second field.
+    get_nano : Integer
+    get_nano = this . internal_zoned_date_time . getNano []
+
+    ## Get the nano-of-second field.
+    get_nanosecond : Integer
+    get_nanosecond = this . get_nano
+
+    ## Get the timezone.
+    get_zone : Zone
+    get_zone = Zone.zone (this . internal_zoned_date_time . getZone [])
+
+    ## Return the number of seconds from the Unix epoch.
+    to_epoch_second : Integer
+    to_epoch_second = this . internal_zoned_date_time . toEpochSecond []
+
+    ## Return the number of milliseconds from the Unix epoch.
+    to_epoch_milli : Integer
+    to_epoch_milli = this . internal_zoned_date_time . toInstant [] . toEpochMilli []
+
+    ## Convert this point in time to time of day.
+    to_time_of_day : Time_Of_Day
+    to_time_of_day = Time_Of_Day.time_of_day (this . internal_zoned_date_time . toLocalTime [])
```

Comment:
@kustosz With this kind of naming, the `time_of_day` function cannot be resolved. I guess the uniform call syntax prevents us from having the `foo = Foo.foo` constructs.

Reply:
Nope, it shouldn't do that. I've checked out your branch and it is true indeed, but it shouldn't be – `to_time_of_day` is not a module-level method, so it should be resolvable through uppercase letters. Renaming the `Time_Of_Day` import helps, but it is a compiler bug.